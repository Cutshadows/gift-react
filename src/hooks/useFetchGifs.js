import { useState, useEffect } from "react"
import { getFetchGifs } from '../helpers/getGifs';

export const useFetchGifs=(category)=>{
    const [state, setState] = useState({
        data:[],
        loading:true
    });

    useEffect(()=>{
         getFetchGifs(category)
                .then(img=>{
                        setState({
                            data:img,
                            loading:false
                        });
                });
    }, [category]) 
    
    
    return state;
}