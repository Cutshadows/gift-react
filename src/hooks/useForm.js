import {useState} from 'react';

export const useForm=(initialState={})=>{
    const [value, setValues] = useState(initialState);

    const reset=()=>{
        setValues(initialState);
    }
    const handleChange=({target})=>{
        setValues({
            ...value,
            [target.name]:target.value
        });
    }

    return [value, handleChange, reset];
}