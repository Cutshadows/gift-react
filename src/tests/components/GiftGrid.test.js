import React from 'react';
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import GiftGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
import '@testing-library/jest-dom';

jest.mock('../../hooks/useFetchGifs');

describe('Pruebas de el componente <GiftGrid></GiftGrid>', () => {
    const category='One Punch';
    
    test('should show the snapshot of the <GiftGrid></GiftGrid>', () => {
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });
        const wrapper=shallow(<GiftGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('should show items when he charges image useFetchGifs', () => {
        const gifs = [{
            id: 'ABC',
            url: 'https://google.com',
            title: 'ABC Title'
        },{
            id: '123',
            url: 'https://google.com',
            title: 'ABC Title'
        }]
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });
        const wrapper=shallow(<GiftGrid category={category}/>);

        
        expect(wrapper).toMatchSnapshot();
        
        expect(wrapper.find('p').exists()).toBe(false)

        expect(wrapper.find('GiftGridItem').length).toBe(gifs.length);

    })
    
    
})
