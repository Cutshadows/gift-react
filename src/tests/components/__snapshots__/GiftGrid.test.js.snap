// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de el componente <GiftGrid></GiftGrid> should show items when he charges image useFetchGifs 1`] = `
<Fragment>
  <li
    className="card animate__animated animate__pulse"
  >
    One Punch
  </li>
  <div
    className="card-grid"
  >
    <GiftGridItem
      id="ABC"
      key="ABC"
      title="ABC Title"
      url="https://google.com"
    />
    <GiftGridItem
      id="123"
      key="123"
      title="ABC Title"
      url="https://google.com"
    />
  </div>
</Fragment>
`;

exports[`Pruebas de el componente <GiftGrid></GiftGrid> should show the snapshot of the <GiftGrid></GiftGrid> 1`] = `
<Fragment>
  <li
    className="card animate__animated animate__pulse"
  >
    One Punch
  </li>
  <p
    className="card animate__animated animate__flash"
  >
    Loading ...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
